ui/src/app/users/components.tsx

```diff
@@ -0,0 +1,32 @@
+import { Dispatch, SetStateAction } from "react"
+
+export type ConfirmationModalData = {
+    onMouseDownFunc: () => void
+    warningText: string
+} | null
+
+interface ConfirmationModalProps {
+    modalData: ConfirmationModalData
+    setModalData: Dispatch<SetStateAction<ConfirmationModalData>>
+}
+
+export function ConfirmationModal({ modalData, setModalData }: ConfirmationModalProps) {
```

Comment:
We should probably specify what is the user that is being deleted in the configmation modal

Reply:
updated